### PR TITLE
Make client cert overridable

### PR DESCRIPTION
**Pull request by shinrich**
This PR is based on PR #4663 which should land first.

This PR allows for proxy.config.ssl.client.cert.filename, proxy.config.ssl.client.private_key.filename, and proxy.config.ssl.client.CA.cert.filename to be overridden by a plugin such as conf_remap.  This override takes precedence over the settings in ssl_server_name.yaml.

The PR updates the documentation and adds tests to exercise remapping all three settings.

### CODE REVIEW

**Comment by shinrich** [comment]
Waiting for fix in PR #4718

**Comment by shinrich** [comment]
Rebased against the merged fix in PR #4718.  Ready to go.

File: iocore/net/SSLNetVConnection.cc [hunk 1]
```diff
@@ -970,7 +971,24 @@ SSLNetVConnection::sslStartHandShake(int event, int &err)
...
+        std::string keyFilePath;
+        if (options.ssl_client_private_key_name) {
+          keyFilePath = Layout::get()->relative_to(params->clientKeyPathOnly, options.ssl_client_private_key_name);
+        }
+        std::string caCertFilePath;
+        if (options.ssl_client_ca_cert_name) {
+          caCertFilePath = Layout::get()->relative_to(params->clientCACertPath, options.ssl_client_ca_cert_name);
+        }
+        clientCTX =
+          params->getCTX(certFilePath.c_str(), keyFilePath.empty() ? nullptr : keyFilePath.c_str(),
```
[the remark this answers is not in the document]
Comment (shinrich) [reply iocore/net/SSLNetVConnection.cc]: Yes, it does potentially load from disk.  We are lazy loading the conf_remap override files.

File: iocore/net/SSLConfig.cc [hunk 2]
```diff
@@ -682,3 +655,84 @@ SSLTicketParams::cleanup()
...
+  ink_mutex_release(&ctxMapLock);
+
+  // Not yet in the table.  Make the cert and add it to the table
+  client_ctx = SSLInitClientContext(this);
+
+  if (client_cert) {
+    // Set public and private keys
+    if (!SSL_CTX_use_certificate_chain_file(client_ctx, client_cert)) {
+      SSLError("failed to load client certificate from %s", client_cert);
+      goto fail;
```
[the remark this answers is not in the document]
Comment (shinrich) [reply iocore/net/SSLConfig.cc]: Following the existing style of the function.

File: iocore/net/SSLConfig.cc [hunk 3]
```diff
@@ -682,3 +655,84 @@ SSLTicketParams::cleanup()
...
 }
+
+SSL_CTX *
+SSLConfigParams::getCTX(const char *client_cert, const char *key_file, const char *ca_bundle_file, const char *ca_bundle_path) const
+{
+  SSL_CTX *client_ctx = nullptr;
+  std::string key;
+  ts::bwprint(key, "{}:{}:{}:{}", client_cert, key_file, ca_bundle_file, ca_bundle_path);
+
+  ink_mutex_acquire(&ctxMapLock);
```
[the remark this answers is not in the document]
Comment (shinrich) [reply iocore/net/SSLConfig.cc]: I acquire, drop, and reacquire the lock if we need to add a new SSL_CTX.  Could add extra scopes to deal with that, but I find acquire/release more straightforward for this use case.

File: iocore/net/SSLNetVConnection.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (shinrich) [reply iocore/net/SSLNetVConnection.cc]: I could add the logic to spawn another thread to do the SSL_CTX load and signal back.

[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply iocore/net/SSLNetVConnection.cc]: The best option might be to toss it over to an `ET_TASK` thread and then have it send an event back to the current thread.

**Pull request merged**